File: benchmarks/tsb/bench_dataframe_apply_map.ts
/**
 * Benchmark: dataFrameApplyMap on 10k-row DataFrame
 */
import { DataFrame, dataFrameApplyMap } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => i * 0.1);
const b = Array.from({ length: ROWS }, (_, i) => i * 0.2);
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++) dataFrameApplyMap(df, (v) => (v as number) + 1);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) dataFrameApplyMap(df, (v) => (v as number) + 1);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "dataframe_apply_map",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dataframe_apply_map.py
"""Benchmark: DataFrame.map element-wise on 10k-row DataFrame"""
import json, time
import pandas as pd

ROWS = 10_000
WARMUP = 3
ITERATIONS = 10
df = pd.DataFrame({"a": [i * 0.1 for i in range(ROWS)], "b": [i * 0.2 for i in range(ROWS)]})

for _ in range(WARMUP):
    df.map(lambda v: v + 1)

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.map(lambda v: v + 1)
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "dataframe_apply_map", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
